Matex/Evaluator.php

```diff
@@ -158,8 +158,7 @@ private function calculate() {
 
 private function perform(string $formula) {
 	$this->pos = 0;
-	if (in_array($formula[0], ['-', '+']))
-		$formula = '0'.$formula;
+	$formula = preg_replace('/(.*[+\-*\/^(])([+\-])([0-9a-z]+)(.*)/i', '${1}(0$2$3)$4', $formula);
```

Comment:
This line is supposed to change any instance of **-#** to **(0-#)** and therefor making it valid an parsable.